### PR TITLE
hexagon: check tensor type when reusing descriptors

### PR DESCRIPTION
**Pull request by w1049**
## Overview

This PR fixes incorrect reuse of Hexagon tensor descriptors exposed by Qwen3-VL-2B.

For a full 512-token ubatch, Qwen3-VL's M-RoPE position input tensor is `I32[2048]`, while the `GET_ROWS` output is `F32[2048, 1]`. Since the hexagon op-batch descriptor lookup currently does not compare `type`, the `F32` output can reuse the `I32` descriptor. `GET_ROWS` then returns `HTP_STATUS_NO_SUPPORT` because `dst->type != HTP_TYPE_F32`, and leaves the inference path with invalid logits.

This change includes a `type` check in `same_shape()`, which is only used in `add_tensor()`.

### Reproduction
Tested on Hexagon v73 and v75.
Using [Qwen3-VL-2B-Instruct Q8_0](https://huggingface.co/Qwen/Qwen3-VL-2B-Instruct-GGUF) on a hexagon device:
```
adb shell "cd /data/local/tmp/llama && \
    LD_LIBRARY_PATH=./lib ADSP_LIBRARY_PATH=./lib \
    ./bin/llama-bench -m /data/local/tmp/gguf/Qwen3VL-2B-Instruct-Q8_0.gguf \
    -p 1024 -n 0 -dev HTP0 -r 1 --no-warmup -v"
```
or
```
adb shell 'cd /data/local/tmp/llama && \
    LD_LIBRARY_PATH=./lib ADSP_LIBRARY_PATH=./lib \
    ./bin/llama-completion -m /data/local/tmp/gguf/Qwen3VL-2B-Instruct-Q8_0.gguf \
    -ngl 99 -dev HTP0 -p "$(printf " test%.0s" $(seq 1016))" -n 32 --temp 0 --no-display-prompt'
```
Before this change, the output contains the error:
```
ggml-hex: HTP0 dspcall : dsp-rsp: NO-SUPPORT
```
And although `llama-completion` produces output, the results do not match the CPU reference.

### Cause
Qwen3-VL-2B has a hidden size of 2048, which happens to align two unrelated dimensions:
- 4 position IDs per token $\times$ 512 (ubatch) tokens = 2048
- hidden size $\times$ one output token = 2048

## Requirements

- I have read and agree with the [contributing guidelines](https://github.com/ggml-org/llama.cpp/blob/master/CONTRIBUTING.md)
- AI usage disclosure: YES. Codex was used to assist with the investigation.

### CODE REVIEW

**Review by max-krasnyansky — APPROVED** [review]
Yep. Good catch from Codex ;-) 
Makes perfect sense.

**Comment by max-krasnyansky** [comment]
@ggml-org/maintainers  need the second approval please

**Review by lhez — APPROVED** [review]
